### PR TITLE
Adding service prompts for protection level

### PR DESCRIPTION
**Pull request by ben-goldberg**
resolves https://github.com/Originate/exosphere-sdk/issues/229

When adding a new service, the user is now prompted to give a protection level, either 'public' or 'private'. This choice is then reflected in the application.yml file.

@kevgo @hugobho @trushton

### CODE REVIEW

**Review by trushton — APPROVED** [review]
Just need to update the feature tests in `exo-create` and it should all work

File: exo-add/features/steps/steps-then.ls [hunk 1]
```diff
@@ -11,7 +11,6 @@ module.exports = ->
 
   @Then /^my application contains the file "([^"]*)" with the content:$/, (file-path, expected-content, done) ->
     fs.read-file path.join(@app-dir, file-path), N (actual-content) ->
-      console.log actual-content.to-string!
```
Comment (kevgo) [inline exo-add/features/steps/steps-then.ls:14]: Thanks for cleaning this up!

File: exo-add/package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "nitroglycerin": "1.1.2",
     "prelude-ls": "1.1.2",
     "tmplconv": "3.2.1",
-    "yaml-cutter": "0.1.3"
+    "yaml-cutter": "^0.2.1"
```
Comment (kevgo) [inline exo-add/package.json]: Please remove the `^` from here.

File: exo-setup/src/cli.ls [hunk 3]
```diff
@@ -14,7 +14,7 @@ module.exports = ->
 
   app-config = yaml.safe-load fs.read-file-sync('application.yml', 'utf8')
   console.log "Setting up #{green app-config.name} #{cyan app-config.version}\n"
-  logger = new Logger flatten [Object.keys(app-config.services[type]) for type of app-config.services]
+  logger = new Logger flatten [Object.keys(app-config.services[type]) unless app-config.services[type] is null for type of app-config.services]
```
Comment (kevgo) [inline exo-setup/src/cli.ls]: We don't care if the services are `null`, `undefined`, or an empty array. Hence, this change can be simplified from `unless app-config.services[type] is null` to `if app-config.services[type]`

**Review by kevgo — APPROVED** [review]
This is cool, thanks!

**Pull request closed without merging**